### PR TITLE
Clean anchor tags to counter XSS

### PR DESCRIPTION
**Pull request by DawoudSheraz**
EDUCATOR-4009

### CODE REVIEW

File: wiki/core/__init__.py [hunk 1]
```diff
@@ -10,5 +11,13 @@ def __init__(self, article, *args, **kwargs):
...
+    return bleach.clean(md.convert(text), strip=True, tags=bleach.sanitizer.ALLOWED_TAGS + ['p'], attributes={
+        'a': is_href_valid
+    })
+
+
+def is_href_valid(tag, name, value):
+    """
+    After mark down, check for any JS inside the value of anchor tag.
+    """
+    return False if value.startswith('javascript:') else True
```
Comment (awaisdar001) [inline wiki/core/__init__.py]: `return not value.startswith('javascript:')`

File: wiki/core/__init__.py [hunk 2]
```diff
@@ -10,5 +11,13 @@ def __init__(self, article, *args, **kwargs):
 
 def article_markdown(text, article, *args, **kwargs):
     md = ArticleMarkdown(article, *args, **kwargs)
-    text = bleach.clean(text, strip=True)
-    return md.convert(text)
+    return bleach.clean(md.convert(text), strip=True, tags=bleach.sanitizer.ALLOWED_TAGS + ['p'], attributes={
```
Comment (awaisdar001) [inline wiki/core/__init__.py]: separate the `text - > html` conversion and cleaning using bleach.

**Comment by awaisdar001** [comment]
please make the pr more descriptive. I have tested it on the sandbox and it looks good to me.

**Review by noraiz-anwar — APPROVED** [review]
👍

**Review by Rabia23 — APPROVED** [review]
Looks good to me 👍

**Review by awaisdar001 — APPROVED** [review]

**Pull request merged**